include/kodlab_mjbots_sdk/pid.h

```diff
@@ -0,0 +1,192 @@
+/**
+ * @file pid.h
+ * @author Chandravaran Kunjeti (kunjeti@seas.upenn.edu)
+ * @brief Class implementation of PID
+ * @date 9/30/2022
+ *
+ * @copyright 2022 The Trustees of the University of Pennsylvania. All rights reserved.
+ *
+ */
+
+#pragma once
+
+#ifndef _PID_
+#define _PID_
+
+#include "Eigen/Geometry"
+
+namespace kodlab
+{
+
+  /**
+   * @brief Pid object
+   */
+  class Pid
+  {
+
+  public:
+    /**
+     * @brief Construct a Pid object
+     * @param p_gain Proportional Gain
+     * @param d_gain Derivative Gain
+     * @param i_gain Integral Gain
+     * @param time_step Time difference between 2 updates
+     */
+    Pid(double p_gain, double d_gain, double i_gain, double time_step, double deadband, double saturation)
+        : kp_(0), kd_(0), ki_(0), time_step_(0.01), deadband_(0), saturation_(0)
+    {
+
+      kp_ = p_gain;
+      kd_ = d_gain;
+      ki_ = i_gain;
+      time_step_ = time_step;
+      deadband_ = deadband;
+      saturation_ = saturation;
```

Comment:
Why are you using an initialization list (with magic number initialization values) and then re-assigning values to these same data members in the constructor body?  This needlessly writes to these members twice (with the initial value immediately discarded).

[This section of the ISO C++ Core Guidelines](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c49-prefer-initialization-to-assignment-in-constructors) indicates that an initialization list is the preferred method here.  But, of course, this should initialize with the passed-in parameter values.

Reply:
I had changed it to this. I had known we could pass the variables we are getting directly. Also when I use the formatting option it makes it a line greater than 80. Any suggestions on fixing this?

Reply:
I gave a solution in my above response...

As for formatting, put a line break between each initializer list element as described in [this style guide section](https://drake.mit.edu/styleguide/cppguide.html#Constructor_Initializer_Lists).

Reply:
Okay figured it out.